src/python/pants/backend/python/lint/pylint/rules.py

```diff
@@ -111,12 +111,21 @@ async def pylint_lint_partition(
     field_set_sources_get = Get(
         SourceFiles, SourceFilesRequest(field_set.source for field_set in partition.field_sets)
     )
-
-    pylint_pex, requirements_pex, prepared_python_sources, field_set_sources = await MultiGet(
+    # Ensure that the empty report dir exists.
+    report_directory_digest_get = Get(Digest, CreateDigest([Directory(REPORT_DIR)]))
```

Comment:
Sounds like this is necessary for Pylint to properly write the report?

Reply:
Yes, it won't create the directory.